### ISSUE TITLE
Password strength indicator 

### ISSUE THREAD

**Issue by benjamincharity** [GetTerminus/terminus-ui#730]

**Comment by brendanjerwin** [comment]
We should talk about this. NIST recommendations have changed. In order to do this, we will actually want a service that can, among other things, check the password against a database of known leaked passwords.

https://pages.nist.gov/800-63-3/sp800-63b.html Section `5.1.1.2 Memorized Secret Verifiers`

**Comment by benjamincharity** [comment]
@brendanjerwin is the thought that we would have 2-part validation of sorts?

1. UI checking that the password input matches our format requirements
2. UI hits service to verify password is not compromised etc.

Would `2` be something that is happening as the user types or something that happens on form submission (like checking that a username is available etc)

**Comment by brendanjerwin** [comment]
something like that. But, keep in mind that format requirements are no longer part of the NIST recommendation. (All that BS around "at least one special character" etc...) 'Strength' is primarily a function of length, lack of repeated characters, and not being in the corpus of known passwords.

**Comment by benjamincharity** [comment]
Makes sense. Will Terminus be updating what we require for passwords? When I started it was a) between 6 & 100, b) contains at least one number.

**Comment by brendanjerwin** [comment]
Yeah. I’m shaking trees to find who I need to talk to. 
BTW I’m noodling in some ideas that might let us do this all client side... thinking we might be able to ship a bloom filter of the compromised password corpus to the client.

**Comment by benjamincharity** [comment]
Oh nice. That sounds like a great idea
